### PR TITLE
feat(mito): Implement SST format for mito2

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
This PR implements the SST format for mito2 engine.

The new SST format encodes the primary keys in a memory-comparable format and stores them as dictionary arrays. We distinguish different time series by comparing the keys of the dictionary array while decoding the `RecordBatch`.

We store three internal columns in parquet:
- `__primary_key`, the primary key of the row (tags).
- `__sequence`, the sequence number of a row.
- `__op_type`, the op type of the row.

The schema of a parquet file is:
```text
field 0, field 1, ..., field N, time index, primary key, sequence, op type
```

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
- #1869

### CODE REVIEW

File: src/mito2/src/error.rs [hunk 1]
```diff
@@ -309,8 +309,28 @@ pub enum Error {
         location: Location,
     },
 
+    #[snafu(display("Failed to get metadata from file {}, reason: {}", file, reason))]
+    NoKeyValue {
```
Comment (waynexia) [inline src/mito2/src/error.rs:313]: the name `NoKeyValue` seems not match the description, do you have a better name?

Comment (evenyag) [reply src/mito2/src/error.rs:313]: I'll change the description. It indicates a key-value pair could not be found in parquet's key value meta.

File: src/mito2/src/sst/parquet/reader.rs [hunk 2]
```diff
@@ -77,7 +94,10 @@ impl ParquetReaderBuilder {
             object_store: self.object_store,
             predicate: self.predicate,
             time_range: self.time_range,
+            projection: self.projection,
             stream: None,
+            read_format: None,
```
[the remark this answers is not in the document]
Comment (evenyag) [reply src/mito2/src/sst/parquet/reader.rs:99]: I'm going to initialize the reader in reader builder then we could remove the `Option`.

**Review by waynexia — APPROVED** [review]
LGTM except the missing tests

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Pull request merged**